src/encoder.rs

```diff
@@ -344,6 +346,53 @@ impl<T: Pixel> FrameState<T> {
     }
   }
 
+  pub fn compute_activity_mask(&mut self) {
+    //activity mask is computed only for luma plane
+
+    let luma_plane = &self.input.planes[0];
+    let PlaneConfig { width, height, .. } = luma_plane.cfg;
+    let luma = &luma_plane.data;
+    let PlaneConfig { xdec, ydec, .. } = self.activity_mask.cfg;
+    let mut act_mask = vec![0 as u32; (width >> xdec) * (height >> ydec) ];
+
+    for r in 0..(height >> ydec) {
+      for c in 0..(width >> xdec) {
+        let base = (r << xdec) * width + (c << ydec);
+
+        //Calculate mean
+        let mut mean: u32 = 0;
+        for x_offset in 0..(1 << xdec) {
+          for y_offset in 0..(1 << ydec) {
+            let offset = (x_offset << xdec) + y_offset;
```

Comment:
Likewise, here `x_offset` need not be shifted as we are iterating the subsampled locations but `y_offset` should be scaled by the luma stride.
```suggestion
            let offset = x_offset + y_offset * stride; // luma_plane.cfg.stride
```

Reply:
Changing his caused `test_encode_decode::tile_encoding_with_stretched_restoration_units_aom` to fail. I'm unsure why.